query/src/main/scala/filodb/query/exec/PromQLGrpcRemoteExec.scala

```diff
@@ -0,0 +1,88 @@
+package filodb.query.exec
+
+import java.util.concurrent.TimeUnit
+
+import io.grpc.Channel
+import io.grpc.stub.StreamObserver
+import kamon.Kamon
+import kamon.trace.Span
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.reactive.{MulticastStrategy, Observable}
+import monix.reactive.subjects.ConcurrentSubject
+
+import filodb.core.DatasetRef
+import filodb.core.query.{PromQlQueryParams, QueryContext, QuerySession}
+import filodb.core.store.ChunkSource
+import filodb.grpc._
+import filodb.grpc.GrpcMultiPartitionQueryService._
+import filodb.query.ProtoConverters._
+import filodb.query.QueryResponse
+
+
+
+trait GrpcRemoteExec extends LeafExecPlan {
+
+
+    def promQlQueryParams: PromQlQueryParams = queryContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+
+    // Since execute of overridden no one will invoke this
+    override def doExecute(source: ChunkSource, querySession: QuerySession)
+                          (implicit sched: Scheduler): ExecResult = ???
+
+    override def execute(source: ChunkSource,
+                         querySession: QuerySession)(implicit sched: Scheduler): Task[QueryResponse] = {
+        val span = Kamon.currentSpan()
+        // Dont finish span since this code didnt create it
+        Kamon.runWithSpan(span, false) {
+            sendGrpcRequest(span, requestTimeoutMs).toListL.map(_.toIterator.toQueryResponse)
+        }
+    }
+
+
+    def args: String = s"${promQlQueryParams.toString}, ${queryContext.plannerParams}, queryEndpoint=$queryEndpoint, " +
+      s"requestTimeoutMs=$requestTimeoutMs"
+
+    def sendGrpcRequest(span: Span, requestTimeoutMs: Long)(implicit sched: Scheduler):
+    Observable[GrpcMultiPartitionQueryService.StreamingResponse]
+
+    def requestTimeoutMs: Long
+
+    def queryEndpoint: String
+
+
+    def getGrpcRequest: Request = {
+        val builder = Request.newBuilder()
+        builder.setQueryParams(promQlQueryParams.toProto)
+        builder.setPlannerParams(queryContext.plannerParams.copy(processMultiPartition = false).toProto)
+        builder.build()
+    }
+
+
+}
+case class PromQLGrpcRemoteExec(channel: Channel,
```

Comment:
Typically ExecPlans are serializable. But `Channel` arg will create problems with serialization of `PromQLGrpcRemoteExec`.
Is it okay to treat as an exception? Or should we think about refactoring this code ?

Reply:
Made this class extend from ``RemoteExec`` for now but ``RemoteExec`` makes assumptions on HTTP as transport and ideally RemoteExec's subclass/trait should be ``HttpRemoteExec`` and Http code should go there. That cleanup can be done later.